samcli/cli/cli_config_file.py

```diff
@@ -166,8 +166,12 @@ def configuration_callback(
     param.default = None
     config_env_name = ctx.params.get("config_env") or DEFAULT_ENV
 
-    config_file = ctx.params.get("config_file") or DEFAULT_CONFIG_FILE_NAME
     config_dir = getattr(ctx, "samconfig_dir", None) or os.getcwd()
+    config_file = (
+        SamConfig.get_default_file(config_dir=config_dir)
+        if getattr(ctx.get_parameter_source("config_file"), "name", "") == "DEFAULT"
```

Comment:
Q: Can this be simplified to something like `ctx.params.get("config_file") if ctx.params.get("config_file") else SamConfig.get_default_file(config_dir=config_dir)`? Just thinking if `confit-file` is provided as input using either `COMMAND_LINE` or `ENVIRONMENT`, it's assumed to not be `DEFAULT`? I checked this for reference https://click.palletsprojects.com/en/8.1.x/advanced/.

Reply:
From what I understand of it, if the value of `--config-file` is not provided, the parameter is set to `DEFAULT`. If it's provided by the user in any other way, it'll be either `COMMAND_LINE` (if provided directly through terminal) or `ENVIRONMENT` (if it's set as an environment variable in the user's terminal). There are some other parts of the click context that may set the config file to the default `samconfig.toml` before this point, and this logic just checks whether that was provided by the user purposefully, or if it's being set to the default. If it's not explicitly set by the user, then we want to first check whether existing `samconfig.yaml` or `samconfig.json` files exist first, and use them if they do exist, otherwise `samconfig.toml`.

Reply:
Can we add a comment to this line so that future readers will know why we're looking at the source and comparing against `DEFAULT`? Additionally, can we compare against the [click.core.ParameterSource](https://click.palletsprojects.com/en/8.1.x/api/#click.core.ParameterSource) enum instead of the name of the `ParameterSource`?